splinepy/io/npz.py

```diff
@@ -10,60 +10,108 @@
 
 import numpy as np
 
+from splinepy import io
+
 
 def load(
     fname,
 ):
     """
-    Read spline in `.npz` form.
+    Read a list of splines in `.npz` form.
 
     Parameters
     -----------
     fname: str
 
     Returns
     --------
-    dict_spline: dict
+    list_spline: list of NURBS / BSpline / Bezier / Rational Bezier
     """
     loaded = np.load(fname)
-    whatami = loaded["whatami"][0]
 
-    dict_spline = {}
-    # update weights
-    if whatami.startswith("NURBS"):
-        dict_spline.update(weights=loaded["weights"])
+    # Initialize an empty list to store the splines
+    list_of_spline_dicts = []
+
+    processed_keys = set()
+
+    # Loop through keys in loaded dictionary and find the ones starting with 'spline_'
+    for key in loaded:
+        if key.startswith("spline_") and key not in processed_keys:
+            # Extract the index of the spline from the key
+            i = int(key.split("_")[1])
+
+            # Initialize an empty dictionary to store the properties of the spline
+            dict_spline = {}
+
+            # Add the common properties of all splines
+            dict_spline["control_points"] = loaded[
+                f"spline_{i}_control_points"
+            ]
+            dict_spline["degrees"] = loaded[f"spline_{i}_degrees"]
+
+            # Add the weights if the spline is rational
+            if f"spline_{i}_weights" in loaded:
+                dict_spline["weights"] = loaded[f"spline_{i}_weights"]
 
-    # update the rest
-    dict_spline.update(control_points=loaded["control_points"])
-    dict_spline.update(degrees=loaded["degrees"])
-    dict_spline.update(knot_vectors=eval(loaded["knot_vectors"][0]))
+            # Add the knot vectors if the spline has them
+            if f"spline_{i}_knot_vectors_0" in loaded:
+                kvs = []
+                for j in range(dict_spline["degrees"].size):
+                    kvs.append(loaded[f"spline_{i}_knot_vectors_{j}"])
+                dict_spline["knot_vectors"] = kvs
 
-    return dict_spline
+            # Append dictionary of spline to the list
+            list_of_spline_dicts.append(dict_spline)
 
+            # Add keys to processed_keys set
+            processed_keys.add(f"spline_{i}_control_points")
+            processed_keys.add(f"spline_{i}_degrees")
+            processed_keys.add(f"spline_{i}_weights")
+            processed_keys.add(f"spline_{i}_whatami")
+            for j in range(dict_spline["degrees"].size):
+                processed_keys.add(f"spline_{i}_knot_vectors_{j}")
 
-def export(fname, spline):
+    return io.ioutils.dict_to_spline(list_of_spline_dicts)
+
+
+def export(fname, list_of_splines):
     """
-    Save spline as `.npz`.
+    Save a list of splines as `.npz`.
 
     Parameters
     -----------
     fname: str
-    spline: BSpline or NURBS
+    splines: list of NURBS / BSpline / Bezier / Rational Bezier
 
     Returns
     --------
     None
     """
-    property_dicts = {
```

Comment:
can you add a check here, if `list_of_splines` is a list? if it is a single spline, you can create a list. Else error.

Reply:
Fixed in [866438a](https://github.com/tataratat/splinepy/pull/280/commits/866438ab03a10e4e3be03aa1f919e4b0c83b42c0)